### ISSUE TITLE
Val loss is very different from training loss when measured on the same data

### ISSUE THREAD

**Issue by jackgle** [tensorflow/tensorflow#59454]
### Issue Type

Bug

### Have you reproduced the bug with TF nightly?

Yes

### Source

source

### Tensorflow Version

2.12.0-dev20221213

### Custom Code

Yes

### OS Platform and Distribution

Linux Ubuntu 18.04

### Python version

3.7, 3.9

### Current Behaviour?

When fine-tuning a Keras Applications model on MNIST, and using the same data for training and validation (for debugging), the validation loss is much higher than the training loss.

P.S.:
I don't believe overfitting is relevant because the training and validation data are the same. I know certain layers like Dropout and BatchNormalization have different behavior during training and validation, but the difference seems too large for this to account for it.

### Standalone code to reproduce the issue

```shell
import numpy as np
np.random.seed(1337)  # for reproducibility
from tensorflow.keras.datasets import mnist
from tensorflow.keras.models import Sequential
import tensorflow.keras.layers as layers
from tensorflow.keras.optimizers import SGD, Adam, RMSprop
import tensorflow as tf

batch_size = 32
nb_classes = 10
nb_epoch = 10

# load data
(X_train, y_train), (X_test, y_test) = mnist.load_data()
del X_test
del y_test

# subsample for memory
idx = np.random.choice(X_train.shape[0], 1000, replace=0)
X_train = X_train[idx]
y_train = y_train[idx]

# preprocess images
tmp = []
for i in range(X_train.shape[0]):
    sample = X_train[i,]
    sample = tf.expand_dims(sample, axis=-1)
    sample = tf.image.grayscale_to_rgb(sample)
    sample = tf.image.resize(sample, (224, 224))
    sample /= 255
    tmp.append(sample)
X_train = np.stack(tmp)

# preprocess labels
Y_train = tf.one_hot(y_train, nb_classes).numpy()

# define model
conv = tf.keras.applications.MobileNetV2(weights='imagenet', include_top=False, input_shape=[224, 224, 3])
for layer in conv.layers:
    layer.trainable = True
inputs = layers.Input((224, 224, 3))
x = conv(inputs)
x = layers.AveragePooling2D((7, 7))(x)
x = layers.Flatten()(x)
outputs = layers.Dense(nb_classes, activation='softmax')(x)
model = tf.keras.models.Model(inputs=inputs, outputs=outputs)

# compile model
model.compile(loss='categorical_crossentropy',
              optimizer='adam')

# train
history = model.fit(X_train, 
                    Y_train,
                    batch_size=batch_size, 
                    epochs=nb_epoch,
                    verbose=1, 
                    validation_data=(X_train, Y_train))
```

### Relevant log output

```shell
Epoch 1/10
32/32 [==============================] - 50s 1s/step - loss: 0.4807 - val_loss: 4.7196
Epoch 2/10
32/32 [==============================] - 29s 912ms/step - loss: 0.1487 - val_loss: 9.1618
Epoch 3/10
32/32 [==============================] - 29s 915ms/step - loss: 0.1485 - val_loss: 15.0174
Epoch 4/10
32/32 [==============================] - 30s 950ms/step - loss: 0.0986 - val_loss: 8.3269
Epoch 5/10
32/32 [==============================] - 30s 925ms/step - loss: 0.0546 - val_loss: 11.8151
Epoch 6/10
32/32 [==============================] - 30s 926ms/step - loss: 0.0379 - val_loss: 11.5285
Epoch 7/10
32/32 [==============================] - 29s 924ms/step - loss: 0.1218 - val_loss: 9.5500
Epoch 8/10
32/32 [==============================] - 29s 906ms/step - loss: 0.0689 - val_loss: 9.0055
Epoch 9/10
32/32 [==============================] - 29s 908ms/step - loss: 0.0452 - val_loss: 6.9891
Epoch 10/10
32/32 [==============================] - 30s 952ms/step - loss: 0.0319 - val_loss: 10.5948
```
</details>

**Comment by jackgle** [comment]
Storing the model weights each epoch, and computing the loss manually for each of them afterwards produces scores equal to what is shown for the `val_loss`. So it is unclear how the training loss is getting so low.

**Comment by jackgle** [comment]
`x = conv(inputs, training=False)` makes the losses approximately equal.

As explained here: https://keras.io/guides/transfer_learning/ 

> When you unfreeze a model that contains BatchNormalization layers in order to do fine-tuning, you should keep the BatchNormalization layers in inference mode by passing training=False when calling the base model. Otherwise the updates applied to the non-trainable weights will suddenly destroy what the model has learned.
